content/blog/2022/2022-01-18-getting-started-with-spring-boot.md

```diff
@@ -0,0 +1,915 @@
+The Hibernate expects that every managed entity has its id set, so it needs to go and call the database to insert the id.
+
+The identity strategy is excellent for fast iteration and the early stages. When we move to the development environment, we should move to something more stable and with better performance.
+
+#### Sequence Generation Strategy
+The `GenerationType.SEQUENCE` strategy benefits from using the sequences inside the databases:
+
+```java
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, 
+            generator = "book_generator")
+    @SequenceGenerator(name = "book_generator",
+            sequenceName = "book_seq",
+            initialValue = 10)
+    private long id;
+}
+```
+To define the sequence generator, we annotate the field with the `@SequenceGenerator`. We declare the name for the Hibernate, the sequence name, and the initial value.
+After defining the generator, we need to connect it to the `@GeneratedValue` annotation by setting its name in the `generator` attribute. 
+
+The sequence strategy uses the sequences from the databases to determine which primary key to chose next.
+
+#### Table Generation Strategy
+Similar to the sequence generator, the table strategy uses the table to keep track of which primary key can be next:
+```java
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.TABLE,
+            generator = "book_generator")
+    @TableGenerator(name = "book_generator", table = "book_id_table")
+    private long id;
+}
+```
+We need to define the `@TableGenerator` annotation with the name and table attributes.
+We provide the generator name to the `generator` attribute inside the `@GeneratedValue` annotation.
+
+The table generator uses
+
+### Defining the Column
+We define the table column with the `@Column` annotation and the name of the column inside the `name` attribute:
+```java
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @Column(name = "title")
+    private String title;
+
+    @Column(name = "author")
+    private String author;
+
+    @Column(name = "publication")
+    private Date publication;
+
+    @Column(name = "numberOfInstances")
+    private int numberOfInstances;
+
+    //The rest of the class omitted
+
+}
+```
+If we do not provide the `name` attribute, the Hibernate will assume that the column name is the same as the variable name inside the Java class.
+
+It is always better to be safe and set the `name` attribute so that things do not start crashing when someone accidentally changes the variable name.
+
+### Defining the ManyToMany Relationship
+
+We define the many-to-many relationship with the `@ManyToMany` annotation
```

Comment:
```suggestion
We define the many-to-many relationship between `Book` and `User` with the `@ManyToMany` annotation
```
It's a good idea to always repeat some context, especially if there were a few sections between introducing a concept and now.